tests/functional/help.sh

```diff
@@ -0,0 +1,140 @@
+nix-env --list-generations --help
+nix-env --delete-generations --help
+nix-env --switch-generation --help
+nix-env --rollback --help
+
+nix-store --help
+nix-store --realise --help
+nix-store --serve --help
+nix-store --gc --help
+nix-store --delete --help
+nix-store --query --help
+nix-store --add --help
+nix-store --add-fixed --help
+nix-store --verify --help
+nix-store --verify-path --help
+nix-store --repair-path --help
+nix-store --dump --help
+nix-store --restore --help
+nix-store --export --help
+nix-store --import --help
+nix-store --optimise --help
+nix-store --read-log --help
+nix-store --dump-db --help
+nix-store --load-db --help
+nix-store --print-env --help
+nix-store --generate-binary-cache-key --help
+
+nix-channel --help
+nix-collect-garbage --help
+nix-copy-closure --help
+nix-daemon --help
+nix-hash --help
+nix-instantiate --help
+nix-prefetch-url --help
+
+nix --help
+nix build --help
+nix bundle --help
+nix copy --help
+nix daemon --help
+nix derivation --help
+nix derivation add --help
+nix derivation show --help
+nix develop --help
+nix doctor --help
+nix edit --help
+nix eval --help
+nix flake --help
+nix flake archive --help
+nix flake check --help
+nix flake clone --help
+nix flake info --help
+nix flake init --help
+nix flake lock --help
+nix flake metadata --help
+nix flake new --help
+nix flake prefetch --help
+nix flake show --help
+nix flake update --help
+nix fmt --help
+nix hash --help
+nix hash file --help
+nix hash path --help
+nix hash to-base16 --help
+nix hash to-base32 --help
+nix hash to-base64 --help
+nix hash to-sri --help
+nix help --help
+nix help-stores --help
+nix key --help
+nix key convert-secret-to-public --help
+nix key generate-secret --help
+nix log --help
+nix nar --help
+nix nar cat --help
+nix nar dump-path --help
+nix nar ls --help
+nix nar pack --help
+nix path-info --help
+nix print-dev-env --help
+nix profile --help
+nix profile diff-closures --help
+nix profile history --help
+nix profile install --help
+nix profile list --help
+nix profile remove --help
+nix profile rollback --help
+nix profile upgrade --help
+nix profile wipe-history --help
+nix realisation --help
+nix realisation info --help
+nix registry --help
+nix registry add --help
+nix registry list --help
+nix registry pin --help
+nix registry remove --help
+nix repl --help
+nix run --help
+nix search --help
+nix shell --help
+nix store --help
+nix store add --help
+nix store add-file --help
+nix store add-path --help
+nix store cat --help
+nix store copy-log --help
+nix store copy-sigs --help
+nix store delete --help
+nix store diff-closures --help
+nix store dump-path --help
+nix store gc --help
+nix store info --help
+nix store ls --help
+nix store make-content-addressed --help
+nix store optimise --help
+nix store path-from-hash-part --help
+nix store ping --help
+nix store prefetch-file --help
+nix store repair --help
+nix store sign --help
+nix store verify --help
+nix upgrade-nix --help
+nix why-depends --help
```

Comment:
This list seems like overkill and a pain to maintain. If we *really* want to test the help of every command, it's better to do something like `nix __dump-cli | jq .args.commands ...` to extract the list of commands.